### PR TITLE
Refactor notifications settings status

### PR DESCRIPTION
**Pull request by alanlanglois**
## Detailed purpose of the PR

[iOS] Fixing the case when user already have the app installed with  access granted for notifications to set suggestion switch "on" by default.

### Testing strategy

- [x] I ensured my code is tested: ✅ test unit ✅ emulator / simulator  
- [ ] I took care to mention breaking changes and extra libs if appropriate

### CODE REVIEW

**Comment by Fausto95** [comment]
Could you constatify the status(just like NOTIFICATION_TYPE)?
Also, shouldn’t we set the suggestion setting to true on home screen did mount?

**Comment by alanlanglois** [comment]
> Could you constatify the status(just like NOTIFICATION_TYPE)?
> Also, shouldn’t we set the suggestion setting to true on home screen did mount?

I'm on it :)

File: src/components/switch.tsx [hunk 1]
```diff
@@ -38,11 +39,12 @@ const styles = StyleSheet.create({
 
 interface Props {
   onPress: () => Promise<void>;
-  isActive: boolean;
+  status: NotificationSettingStatus;
```
Comment (alanlanglois) [inline src/components/switch.tsx:42]: should stay is active IMO, as switch could be used in other case not involving notifications

Comment (Fausto95) [reply src/components/switch.tsx:42]: Agreed

Comment (Fausto95) [reply src/components/switch.tsx:42]: So we need to change from `status` to `isActive` here: https://github.com/CoorpAcademy/mobile/blob/31b6944099b2d71d247d0ae8f2f6324c48278ed5/src/screens/settings.tsx#L41

Comment (alanlanglois) [reply src/components/switch.tsx:42]: Are you sure?
I just changed this in the component:
`<Switch isActive={item.status === NOTIFICATION_SETTINGS_STATUS.ACTIVATED}
`
So we can keep a String argument for analytics

File: src/const.ts [hunk 2]
```diff
@@ -169,6 +170,15 @@ export const RESOURCE_MIME_TYPE: Record<
   JPEG: 'image/jpeg',
 };
 
+export const NOTIFICATION_SETTINGS_STATUS: Record<
+  'IDDLE' | 'ACTIVATED' | 'DESACTIVATED',
+  NotificationSettingStatus
+> = {
```
Comment (Fausto95) [inline src/const.ts:176]: IDLE

File: src/const.ts [hunk 3]
```diff
@@ -169,6 +170,15 @@ export const RESOURCE_MIME_TYPE: Record<
...
+  IDDLE: 'idle',
+  ACTIVATED: 'activated',
+  DESACTIVATED: 'deactivated',
```
Comment (Fausto95) [inline src/const.ts:179]: DEACTIVATED

**Review by Fausto95 — CHANGES_REQUESTED** [review]

File: src/const.ts (hunk 3 above)
Comment (alanlanglois) [reply src/const.ts:179]: Excuse ma French...

**Review by Fausto95 — APPROVED** [review]

**Pull request merged**